### ISSUE TITLE
Breaks JS on frontend in IE11

### ISSUE THREAD

**Issue by adrianbj** [BernhardBaumrock/FieldtypeRockGrid#3]
Hey @BernhardBaumrock - this line: https://github.com/BernhardBaumrock/FieldtypeRockGrid/blob/cc8115d41150754bc7e07928ff46c1182ca25ecc/FieldtypeRockGrid.module.php#L42

gets output on the frontend of your site and breaks other JS in IE11.

Surely this should be limited to the backend and instances where RockGrid is actually displayed? Also, I expect it will break the PW in the backend on IE11 as well.

Let me know if you need any other info.

**Comment by BernhardBaumrock** [comment]
Thank you @adrianbj for reporting this!
Of course you are right that it should only be injected when a RockGrid is displayed somewhere on the page. I know that this does not work properly yet. I think some other files are also loaded even when no RockGrid is loaded. I think this happened because it's sometimes tricky to have all files in place when things are requested via AJAX, but not sure any more.
No idea about IE11. I don't need to support IE11 on my running installations and for the future I think I will use RockTabulator. I want to give it a try soon and see how it goes. This could also replace my HandsonTable field that is not MIT any more from version 7 on: https://processwire.com/talk/topic/16608-fieldtypehandsontable-excel-like-inputfield/?do=findComment&comment=183295

A quickfix would be to simply make the hook only load in the pw admin:
```php
    $this->wire->addHookAfter('Page(template=admin)::render', function($event) {
      ...
    });
```

I think it would not really be a problem to drop "support" of RockGrid in the frontend. I don't know of anybody that has ever used it on the frontend. Some plugins that I've created even don't work on the frontend - I didn't have the need and time to support it.

**Comment by adrianbj** [comment]
Yeah, that would probably be fine - I don't think backend code needs to support IE11, but the problem is that until Microsoft ports Edge to Windows 7 (which is coming), any code that completely halts execution of JS scripts (like that use of dispatchEvent does), needs to not be run on the frontend.

Cheers!
